caom2-compute/src/main/java/ca/nrc/cadc/caom2/compute/CaomWCSValidator.java

```diff
@@ -303,64 +291,121 @@ public static void validatePolarizationWCS(String context, PolarizationWCS polar
         }
     }
 
-    public static void validateAxes(Chunk chunk)  {
-        // Keep track of all errors in the axis definition
-        String errorMsg = "";
+    private static void checkDuplicateAxis(String[] axisList, Integer axis, String varName) {
+        log.debug("checking duplicate: " + axis);
+        if (axisList[axis] != null) {
+            throw new IllegalArgumentException("Duplicate axis found: (" + axis + ") " + axisList[axis] + " & " + varName);
+        }
+    }
 
+    public static void validateAxes(Chunk chunk)  {
         if (chunk.naxis != null) {
             // Have axisList offset by 1 because the list will be counted
-            // from 1 to naxis. Nulls in the list are missing axi
-            // definitions.
+            // from 1 to naxis. Nulls in the list are missing axis definitions.
             String[] axisList = new String[chunk.naxis + 1];
 
-            Class myObjectClass = Chunk.class;
-
-            Field[] fields = myObjectClass.getFields();
-            Integer naxis = chunk.naxis;
-
-            for (Field f : fields) {
-                if (f.getType().getSimpleName().compareTo("Integer") == 0) {
-                    String fieldName = f.getName();
-
-                    // While building the axisList, duplicate entries can be found
-                    if (fieldName.compareTo("naxis") != 0) {
-                        try {
-                            Integer val = (Integer) f.get(chunk);
-                            if (val != null) { // Ignore axes that are not defined as part of the metadata
-                                if (val <= naxis) { // Ignore axes greater than naxis: situation is allowed
-                                    if (axisList[val] != null) { // Flag duplicate axis definitions
-                                        errorMsg += "\tDuplicate axis number: " + val + ": " + fieldName + ", " + axisList[val] + ".";
-                                    } else {
-                                        axisList[val] = fieldName;
-                                    }
-                                }
-                            }
-                        } catch (IllegalAccessException iae) {
-                            // This error is thrown by the f.get on line 322
-                            throw new IllegalArgumentException(AXES_VALIDATION_ERROR + ": could not access Chunk to validate axes.");
-                        }
-                    }
+            // Go through each axis and validate
+
+            // positionAxis1 and positionAxis2 must both be defined. If they are, position metadata must be defined
+            if (chunk.positionAxis1 != null && chunk.positionAxis1 <= chunk.naxis) {
```

Comment:
The comment here is wrong: saying "must both be defined" makes them mandatory which is not true. The docs do not say anything about the positionAxis1|2 having anything to do with naxis -- the conditions apply even if the value(s) are greater than naxis. 

I think this can more concisely be written as a single block starting with:
```if (chunk.positionAxis1 != null || chunk.positionAxis2 != null) {```
which means detailed axis index checking required (eg check that three things are all non-null). Doing it this way also means getting rid of duplicate code (eg lines ~316 and ~334.

Side note: you do not call checkDuplicateAxes before adding positionAxis1 to the array (technically unnecessary since it is currently the first one) but then you also don't call it before adding positinoAxis2 (which means this doesn't handle the case where they are equal). It would be better to put the checkDuplicateAxes call before both of them -- the first one that can't actually fail is preferrable because it is less subtle and if someone later on re-orders the checks it won't introduce a subtle bug. Aside: I would order them position, energy, time, pol, custom but someone else might order then alphabetically, and you have time before energy for a reason... there is no "right" order.